src/Maker/Factory/MakeFactoryData.php

```diff
@@ -62,6 +65,14 @@ public function __construct(
         }
 
         $this->methodsInPHPDoc = $withPhpDoc ? MakeFactoryPHPDocMethod::createAll($this) : [];
+        $reflectionExtractor = new ReflectionExtractor();
+        $this->propertyInfo = new PropertyInfoExtractor(
+            [],
+            [],
+            [],
+            [$reflectionExtractor],
+            [$reflectionExtractor]
+        );
```

Comment:
few things about this: 

- I'm not opposed to use property info component (I do prefer this solution to the `newInstanceWithoutConstructor` stuff), but we should make the dependency explicit in `composer.json` 
- I think we can only use `ReflectionExtractor`
- please, store this in a static property:
```php
private static (PropertyAccessExtractorInterface&PropertyInitializableExtractorInterface)|null $propertyInfo;

private static function propertyInfo(): PropertyAccessExtractorInterface&PropertyInitializableExtractorInterface
{
    return self::$propertyInfo ??= new ReflectionExtractor();
}
```

Reply:
My IDE is screaming at me that that union type is a 8.2 feature but minimum version is 8.1 :)

Reply:
and your IDE is right :)

well, actually not union type, but "[Disjunctive Normal Form Types](https://wiki.php.net/rfc/dnf_types)"

just use `ReflectionExtractor` everywhere :shrug:

Reply:
It's actually working as return type but not as variable type apparently, I just annotated it and set it to mixed for the variable. Seems to work ok

Reply:
Yes, the return type is not a "DNF" (the ones using union and intersection, but just simple intersection